elasticsearch-util/src/test/java/org/sunbird/common/ElasticSearchUtilTest.java

```diff
@@ -4,687 +4,902 @@
+		physicsMap = intitializePhysicsCourse(60);
+		
+
+	}
+
+	@Before
+	public void initBeforeTest()  {
+		
+		//Initialize mock objects before the tests.
+		mockBaseRules();
+		mockRulesForGet();
+		mockRulesForInsert();
+		mockRulesForUpdate();
+		mockRulesForDelete();
+		
+		mockRulesForIndexes();
+		mockRulesHttpRequest();
+		mockRulesBulkInsert();
+
+	}
+	
+	@Test
+	public void testConnectionSuccess() {
+		TransportClient client = mock(TransportClient.class);
+		assertNotNull(client);
+	}
+
+	@Test
+	public void testCreateIndexSuccess() {
+		boolean response = ElasticSearchUtil.createIndex(INDEX_NAME, TYPE_NAME, ElasticSearchMapping.createMapping(),
+				ElasticSearchSettings.createSettingsForIndex());
+		try {
+			Thread.sleep(2000);
+		} catch (InterruptedException e) {
+			ProjectLogger.log("Index creation time out");
+		}
+		assertTrue(response);
+	}
+
+	@Test
+	public void testCreateDataSuccess() {
+
+		String responseId = ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) chemistryMap.get("courseId"), chemistryMap);
+		// inserting second record
+		ElasticSearchUtil.createData(INDEX_NAME, TYPE_NAME, (String) physicsMap.get("courseId"), physicsMap);
+		assertEquals(responseId, chemistryMap.get("courseId"));
+		// wait for 1 second since the created data will reflect after some time
+		try {
+			Thread.sleep(1000);
+		} catch (InterruptedException e) {
+			ProjectLogger.log(e.getMessage(), e);
+		}
+	}
+
+	@Test
+	public void testGetByIdentifierSuccess() {
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME,
+				(String) chemistryMap.get("courseId"));
+		assertEquals(responseMap.get("courseId"), chemistryMap.get("courseId"));
+	}
+
+	@Test
+	public void testUpdateByIdentifierSuccess() {
+		Map<String, Object> innermap = new HashMap<>();
+		innermap.put("courseName", "updatedCourese name");
+		innermap.put("organisationId", "updatedOrgId");
+		
+		GetRequestBuilder grb = mock(GetRequestBuilder.class);
+		GetResponse getResponse = mock(GetResponse.class);
+		when(client.prepareGet(Mockito.anyString(), Mockito.anyString(), Mockito.eq((String) chemistryMap.get("courseId"))))
+				.thenReturn(grb);
+		when(grb.get()).thenReturn(getResponse);
+		when(getResponse.getSource()).thenReturn(innermap);
+		
+		
+		boolean response = ElasticSearchUtil.updateData(INDEX_NAME, TYPE_NAME, (String) chemistryMap.get("courseId"), innermap);
+		assertTrue(response);
+		Map<String, Object> responseMap = ElasticSearchUtil.getDataByIdentifier(INDEX_NAME, TYPE_NAME,
+				(String) chemistryMap.get("courseId"));
+		assertEquals(responseMap.get("courseName"), "updatedCourese name");
+	}
+
+	@Test
+	public void testComplexSearchSuccess() throws Exception {
+		SearchDTO searchDTO = new SearchDTO();
+		List<String> fields = new ArrayList<String>();
+		fields.add("courseId");
+		fields.add("courseType");
+		fields.add("createdOn");
+		fields.add("description");
+		List<String> excludefields = new ArrayList<String>();
```

Comment:
excludeFields

Reply:
Did not understand? what action is required.